src/i_truecolor.c

```diff
@@ -37,6 +37,100 @@ typedef union
+const uint32_t (*I_BlendOverFunc) (const uint32_t bg_i, const uint32_t fg_i, const int amount);
+const uint32_t (*I_BlendOverAltFunc) (const uint32_t bg_i, const uint32_t fg_i, const int amount);
+
+// [JN] Different blending alpha values for different games
+#define OVERLAY_ALPHA_TRANMAP     0xA8  // Doom: TRANMAP, 168 (66% opacity)
+#define OVERLAY_ALPHA_TINTTAB     0x60  // Raven: TINTTAB, 96 (38% opacity)
+#define OVERLAY_ALPHA_TINTTABALT  0x8E  // Raven: TINTTAB, 142 (56% opacity, "Alt")
+#define OVERLAY_ALPHA_XLATAB      0xC0  // Strife: XLATAB, 192 (75% opacity)
+#define OVERLAY_ALPHA_XLATABALT   0x40  // Strife: XLATAB, 64 (25% opacity, "Alt")
+
+
+// [JN] Initialize blending maps for tablified additive and overlay translucency
+void R_InitBlendMaps (GameMission_t mission)
+{
+    tcpixel_t bg, fg;
+    tcpixel_t retAdd, retOver;
+    uint8_t overlay_alpha, overlay_alt_alpha;
+
+    // Define different alpha values for different games
+    switch (mission)
+    {
+        default: // Doom and derivatives
+        {
+            overlay_alpha = OVERLAY_ALPHA_TRANMAP;
+            overlay_alt_alpha = 0; // "alt" blending is not used
+            break;
+        }
+
+        case heretic:
+        case hexen:
+        {
+            overlay_alpha = OVERLAY_ALPHA_TINTTAB;
+            overlay_alt_alpha = OVERLAY_ALPHA_TINTTABALT;
+            break;
+        }
+
+        case strife:
+        {
+            overlay_alpha = OVERLAY_ALPHA_XLATAB;
+            overlay_alt_alpha = OVERLAY_ALPHA_XLATABALT;
+            break;
+        }
+    }
+
+    // Shortcut: these variables are always same in tablified approach
+    retAdd.a = 0xFFU;
+    retOver.a = 0xFFU;
+
+    for (int bg_index = 0; bg_index < 512; ++bg_index)
+    {
+        for (int fg_index = 0; fg_index < 512; ++fg_index)
+        {
+            // Convert LUT indices back to RGB values with reduced bit depth
+            bg.r = (bg_index >> 6) << 5;
+            bg.g = ((bg_index >> 3) & 0x07) << 5;
+            bg.b = (bg_index & 0x07) << 5;
+
+            fg.r = (fg_index >> 6) << 5;
+            fg.g = ((fg_index >> 3) & 0x07) << 5;
+            fg.b = (fg_index & 0x07) << 5;
+
+            // Additive blending calculation
+            retAdd.r = MIN(bg.r + fg.r, 0xFFU);
+            retAdd.g = MIN(bg.g + fg.g, 0xFFU);
+            retAdd.b = MIN(bg.b + fg.b, 0xFFU);
+            blendAddLUT[bg_index][fg_index] = retAdd.i;
+
+            // Overlay blending calculation
+            retOver.r = (overlay_alpha * fg.r + (0xFFU - overlay_alpha) * bg.r) >> 8;
+            retOver.g = (overlay_alpha * fg.g + (0xFFU - overlay_alpha) * bg.g) >> 8;
+            retOver.b = (overlay_alpha * fg.b + (0xFFU - overlay_alpha) * bg.b) >> 8;
+            blendOverLUT[bg_index][fg_index] = retOver.i;
+
+            // Overlay "alt" blending calculation
+            retOver.r = (overlay_alt_alpha * fg.r + (0xFFU - overlay_alt_alpha) * bg.r) >> 8;
```

Comment:
You could check for `overlay_alt_alpha == 0` and generate the additive values only then.